client/state/ui/first-view/selectors.js

```diff
@@ -49,3 +58,18 @@ export function shouldViewBeVisible( state ) {
 		switchedFromDifferentSection( state ) &&
 		! isSectionLoading( state );
 }
+
+function isLikelyInSafariPrivateMode() {
+	// Safari doesn't allow access to Local Storage when in Private Mode;
+	// We will use this check as a proxy to determine whether or not Safari
+	// is in Private Mode
+	let isAbleToAccessLocalStorage = true;
+
+	try {
+		localStorage.checkIfFirstViewPreferencesCanBeCached = null;
```

Comment:
@ehg We will pull this back out of the PR and tackle it in a separate PR where we can discuss more and come to the best solution, taking into account things like SSR. Thanks for pointing this out and questioning it!